src/main/kotlin/ai/digital/integration/server/common/cluster/operator/OperatorHelper.kt

```diff
@@ -444,32 +454,95 @@ abstract class OperatorHelper(val project: Project, val productName: ProductName
         return productName.toString().toLowerCase()
     }
 
-    fun cleanUpCluster() {
-
-        project.logger.lifecycle("Clean up cluster resources")
+    fun cleanUpCluster(waiting: Duration) {
 
-        val productNames = ProductName.values()
-        val resources = arrayOf(
+        val resourcesList1 = arrayOf(
             "all",
             "roles",
             "roleBinding",
             "clusterRoles",
             "clusterRoleBinding",
+            "ing",
             "ingressclass",
-            "crd",
             "pvc"
         )
+        // repeat delete of following resources to be sure that all is clean
+        val resourcesList2 = arrayOf(
+            "service",
+            "crd"
+        )
+
+        for (iteration in 1..2) {
```

Comment:
Better to use coroutine here, code will be slightly cleaner and a bit more high level